### DISCUSSION TITLE
Why remove CM v4.0 mappings?

### DISCUSSION THREAD

**Discussion by joaoserra** [nekno/DV7toDV8#5] [Q&A] [answered]
When converting to DV8 BL+RPU from DV7 BL+EL+RPU, i understand why polynomial/MMR mappings must be removed, but why remove CMv4 data and only keep CMv2.9? What's the rationale? 
Thanks!

**Comment by Brandonshire** [comment]
I'm also curious about this. I'm just learning about all of this stuff, so there may be things I don't understand, but is there an advantage to removing CMv4 data?

**Comment by nekno** [comment, marked as answer]
Hi @joaoserra and @Brandonshire — sorry I missed this discussion in December.

The short answer is that CMv4.0 data is stripped to make the files playable with DV on Apple TV.

The Apple TV is a CMv4.0 device, but up until tvOS 17, that only applied to DV5 (ITP BL) and DV8.4 (HLG BL) content.

In tvOS 17, Apple added support for native DV8.1 (HDR10 BL) playback, but currently, the only app that seems to use Apple's native AV player is the 1st-party Computers app, which can play back mp4 or mov files with DV8.1 content when shared from your computer's iTunes library with Home Sharing.

This DV7 to DV8 app was created to accommodate the limitations of the Apple TV and tvOS 16, to get DV Profile 8.1 working in the [Infuse app made by Firecore](https://firecore.com).

Currently, Infuse is the only 3rd-party app that will play files with proper DV8.1 support, using their own proprietary approach, but it's limited to CMv2.9 metadata, even on tvOS 17.

Plex can play DV5 files by switching over to Apple's native player, but it doesn't yet do so for DV8 files. A DV8 file will play back using their own player and trigger the DV mode on your TV, but it won't actually render/send the DV dynamic metadata (RPU) — you'll just be watching content in HDR10 with your TV in DV mode. This is known as "fake DV".

Firecore is currently working on support for DV8 with CMv4.0, presumably by switching to Apple's native AV player for those files, but their approach is TBD. Perhaps they can alternatively support CMv4.0 using their own player on tvOS 17. 

Firecore just released Infuse 7.7. The next update, 7.7.1, is planned to include support for DV8.1 with CMv4.0 or at least native playback support.

So, if everything goes to plan, this app can be changed to stop removing CMv4.0 metadata, and we should go back and re-convert our files.

**Reply by Brandonshire** [reply]
Thank you so much for your detailed answer. I have just started to learn about a lot of this stuff, after long thinking anything that was profile 7 was just doomed to only be playable on an AppleTV in HDR 10, so I've been trying to learn as much about converting files to profile 8 as I can, and only just came across this as an option. It was actually in reading about recent versions of Infuse that I learned about this as an option. After I had left this comment I actually came across some threads on the Firecore forums and had started to piece together the reasoning, though your explanation is much clearer than reading through hundreds of posts and trying to fit it all together (I actually came across some posts of yours there a short time after your answer here). Anyway thanks, this makes sense and I'm currently using your tool to convert a few of my movies with MEL layers to see how they play in the current version of Infuse. Here's hoping some time in the next few versions we'll be able to get even better playback!

Thanks again for your answer and for your wonderful tool!
